connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/DistributedHerderTest.java

```diff
@@ -935,9 +936,6 @@ public Object answer() throws Throwable {
         herder.tick();
         assertTrue(connectorConfigCb.isDone());
         assertEquals(CONN1_CONFIG_UPDATED, connectorConfigCb.get());
-        // The config passed to Worker should
```

Comment:
I guess its ok that this gets removed, but seems a bit weird. This basically means we have parsed with the wrong config class. It ends up not mattering because the worker doesn't pass the parsed config directly to the connector, it grabs the originals and lets the connector parse it, but this might be another reason for having the `SourceConnectorConfig` class and making `ConnectorConfig` abstract so it cannot be used incorrectly.

Let me know if that makes sense or what the potential fallout is -- I'm not sure if making that change will end up leaking all over the code.